File: samples/client/petstore/dart/lib/auth/api_key_auth.dart
part of api;

class ApiKeyAuth implements Authentication {

  final String location;
  final String paramName;
  String apiKey;
  String apiKeyPrefix;

  ApiKeyAuth(this.location, this.paramName);

  @override
  void applyToParams(Map<String, String> queryParams, Map<String, String> headerParams) {
    String value;
    if (apiKeyPrefix != null) {
      value = '$apiKeyPrefix $apiKey';
    } else {
      value = apiKey;
    }

    if (location == 'query' && value != null) {
      queryParams[paramName] = value;
    } else if (location == 'header' && value != null) {
      headerParams[paramName] = value;
    }
  }

}

File: samples/client/petstore/kotlin-multiplatform-new/src/common/main/org/openapitools/client/auth/ApiKeyAuth.kt
package org.openapitools.client.auth

import org.openapitools.client.infrastructure.Queries

@Suppress("MemberVisibilityCanBePrivate")
public class ApiKeyAuth(public val location: String, public val paramName: String) : Authentication {
    public var apiKey: String? = null
    public var apiKeyPrefix: String? = null

    override fun apply(queries: Queries, headers: MutableMap<String, String?>) {
            val key: String = apiKey ?: return
            val prefix: String? = apiKeyPrefix
            val value: String = if (prefix != null) "$prefix $key" else key
            when (location) {
                "query" -> queries.add(paramName, value)
                "header" -> headers[paramName] = value
            }
        }
}
